api/v1alpha1/clusterconfig_types.go

```diff
@@ -186,6 +191,40 @@ func (ExtraAPIServerCertSANs) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
+// ImageRegistryCredentials required for providing credentials for an image registry URL.
+type ImageRegistryCredentials struct {
```

Comment:
I've kept this struct independent from the related Containerd mirrors configuration.

Reply:
Great idea!

Reply:
Now I don't know anymore lol 🤔 

This PR adds the following:
```
  matchImages:
  - {{ printf "%q" .MirrorHost }}
  - "*"
  - "*.*"
  - "*.*.*"
  - "*.*.*.*"
  - "*.*.*.*.*"
  - "*.*.*.*.*.*"
```
https://github.com/d2iq-labs/capi-runtime-extensions/pull/174/files#diff-3996a5b712640bdb338abbda2b49ab7519154b412bdb3072be1c65e5d8917dccR13-R18

But that only makes sense if the registry is also a mirror.

With https://github.com/d2iq-labs/capi-runtime-extensions/pull/174#discussion_r1335902058 its also getting pretty difficult to reason what all these configurations are for and how to use globs as URLs or try to match up a glob to a registry mirror.

Maybe we want something like this instead?:
```
imageRegistries:
  credentials:
  - url: [https://]myregistry.com # intentionally a URL with a potential schema and not an image glob
    username: <>
    password: <>
    mirror: false
```

Reply:
These credentials should only be in the dynamic-credential-provider config file though right? And the URL/glob could map to a single username/password so I think this is OK. I don't think mirror makes any difference here.

Reply:
Discussed this offline, to keep it simple at first, we will ask for a URL with the credentials.
Then when image registry mirrors are implemented, we will use the URL to also configure Containerd and the DynamicCredentialProvider configuration as a mirror.